PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -302,7 +302,17 @@ class FileDialogEventHandler : public IFileDialogEvents, public IFileDialogContr
 		_currentType = dialogIndex;
 		generic_string name = getDialogFileName(_dialog);
 		if (changeExt(name, dialogIndex - 1))
-			return SUCCEEDED(_dialog->SetFileName(name.c_str()));
+		{
+			// Set file name and restore the previous selection in the edit box.
+			// The selection is modified by SetFileName().
+			DWORD selStart = 0;
+			DWORD selEnd = 0;
+			SendMessage(_hwndNameEdit, EM_GETSEL, reinterpret_cast<WPARAM>(&selStart), reinterpret_cast<LPARAM>(&selEnd));
```

Comment:
`selStart` & `selEnd` are always 0 after calling EM_GETSEL.
I also tried the following codes, the same result:
```cpp
			long res = SendMessage(_hwndNameEdit, EM_GETSEL, reinterpret_cast<WPARAM>(&selStart), reinterpret_cast<LPARAM>(&selEnd));
			selStart = LOWORD(res);
			selEnd = HIWORD(res);
```
_hwndNameEdit is not null and I have no idea why it doesn't work on my Windows11.

Reply:
Done.